remote-api/src/main/java/org/alfresco/repo/web/scripts/content/ContentGet.java

```diff
@@ -140,6 +148,22 @@ public void execute(WebScriptRequest req, WebScriptResponse res)
                 propertyQName = QName.createQName(propertyName, namespaceService);
             }
         }
+        // determine attachment and force download for specific mimetypes - see PRODSEC-5862
+        boolean attach = Boolean.valueOf(req.getParameter("a"));
+        ContentReader reader = contentService.getReader(nodeRef, propertyQName);
+        String mimetype = MimeTypeUtil.determineMimetype(reader,req,mimetypeService);
+
+        if (!attach)
+        {
+            if ((nonAttachContentTypes != null) && (!(nonAttachContentTypes.contains(mimetype))))
+            {
+                attach = true;
+            }
+            else
+            {
+                logger.warn("Ignored a=false for " + nodeRef.getId() + " since " + mimetype + " is not in the whitelist for non-attach content types");
+            }
```

Comment:
This else block won't be needed after the if conditional is properly inverted. The code should set override `attach = true` and print the log within the same conditional block, it doesn't make sense for this statement to be here.